app/core/ai_monitoring.py

```
@@ -0,0 +1,186 @@
+import threading
+from dataclasses import dataclass
+import json
+
+@dataclass
+class MetricPoint:
+    timestamp: datetime
+    value: float
+    context: Dict
+
+class AIResourceMonitor:
+    def __init__(self, db_path: str = 'ai_metrics.db'):
```

Ruff (0.8.2)

Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)